CIScripts/Test/TestConfigurationUtils.ps1

```diff
@@ -203,69 +205,71 @@ function Disable-AgentService {
+        Write-Host "Installing Agent"
+        Start-Process msiexec.exe -ArgumentList @("/i", "C:\Artifacts\contrail-vrouter-agent.msi", "/quiet") -Wait
 
-        Start-Sleep -s $TimeBetweenChecksInSeconds
+        # Refresh Path
+        $Env:Path = [System.Environment]::GetEnvironmentVariable("Path", "Machine")
     }
-
-    throw "Agent service is not enabled. EXPECTED: Agent service is enabled"
 }
 
-function Assert-IsAgentServiceDisabled {
-    Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session)
+function Uninstall-Agent {
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session)
 
-    $MaxWaitTimeInSeconds = $MAX_WAIT_TIME_FOR_AGENT_IN_SECONDS
-    $TimeBetweenChecksInSeconds = $TIME_BETWEEN_AGENT_CHECKS_IN_SECONDS
-    $MaxNumberOfChecks = [Math]::Ceiling($MaxWaitTimeInSeconds / $TimeBetweenChecksInSeconds)
-
-    for ($RetryNum = $MaxNumberOfChecks; $RetryNum -gt 0; $RetryNum--) {
-        $Status = Invoke-Command -Session $Session -ScriptBlock {
-            return $((Get-Service "ContrailAgent" -ErrorAction SilentlyContinue).Status)
-        }
-        if ($Status.Value -eq "Stopped") {
-            return
-        }
+    Invoke-Command -Session $Session -ScriptBlock {
+        Write-Host "Uninstalling Agent"
+        Start-Process msiexec.exe -ArgumentList @("/x", "C:\Artifacts\contrail-vrouter-agent.msi", "/quiet") -Wait
 
-        Start-Sleep -s $TimeBetweenChecksInSeconds
+        # Refresh Path
+        $Env:Path = [System.Environment]::GetEnvironmentVariable("Path", "Machine")
     }
-
-    throw "Agent service is not disabled. EXPECTED: Agent service is disabled"
 }
 
-function Assert-AgentProcessCrashed {
-    Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session,
-           [Parameter(Mandatory = $true)] [DateTime] $After,
-           [Parameter(Mandatory = $false)] [Int] $TimeoutSeconds = 60)
-
-    $TimeBetweenChecksInSeconds = 2
-
-    foreach ($i in 0..($TimeoutSeconds / $TimeBetweenChecksInSeconds)) {
-        $Res = Invoke-Command -Session $Session -ScriptBlock {
-            Get-EventLog -LogName "System" -EntryType "Error" -Source "Service Control Manager" `
-                -Message "The ContrailAgent service terminated unexpectedly*" -After ($Using:After).addSeconds(-1)
-        }
+function Get-AgentServiceStatus {
+    Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session)
+    $Status = Invoke-Command -Session $Session -ScriptBlock {
+        (Get-Service "ContrailAgent").Status
+    }
+    return $Status.Value
+}
 
-        if ($Res) {
-            return
-        }
+function Assert-IsAgentServiceEnabled {
+    Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session)
+    $Status = Retry-Command -IntervalInSecs $TIME_BETWEEN_AGENT_CHECKS_IN_SECONDS `
+                            -TimeoutInSecs $MAX_WAIT_TIME_FOR_AGENT_IN_SECONDS `
+                            { Get-AgentServiceStatus -Session $Session }
+    if ($Status -eq "Running") {
```

Comment:
I think it has different semantics now vs. the original – it won't work in a case when agent service is "Stopped" for a moment.

Reply:
You mean that

`throws -> throws -> throws -> Running`
`throws -> throws -> Stopped -> Running`

in the second case, `Retry-Command` will return `Stopped` prematurely? And old implementation would handle it properly?

Makes sense. Will fix